packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -24,6 +26,67 @@ class DBot {
     async initWorkspace(public_path, store, api_helpers_store, is_mobile) {
         const recent_files = await getSavedWorkspaces();
 
+        api_base.init();
+        this.interpreter = Interpreter();
+        const that = this;
+        Blockly.Blocks.trade_definition_tradetype.onchange = function (event) {
+            if (!this.workspace || this.isInFlyout || this.workspace.isDragging()) {
+                return;
+            }
+
+            this.enforceLimitations();
+
+            if (event.type === Blockly.Events.BLOCK_CHANGE) {
+                if (event.name === 'SYMBOL_LIST' || event.name === 'TRADETYPECAT_LIST') {
+                    const { contracts_for } = ApiHelpers.instance;
+                    const top_parent_block = this.getTopParent();
+                    const market_block = top_parent_block.getChildByType('trade_definition_market');
+                    const market = market_block.getFieldValue('MARKET_LIST');
+                    const submarket = market_block.getFieldValue('SUBMARKET_LIST');
+                    const symbol = market_block.getFieldValue('SYMBOL_LIST');
+                    const trade_type_cat = this.getFieldValue('TRADETYPECAT_LIST');
+                    const trade_type = this.getFieldValue('TRADETYPE_LIST');
+
+                    if (event.name === 'SYMBOL_LIST') {
```

Comment:
`event.name` is being used multiple times in this scope. Can we de-structure it from `event` instead?